src/main/java/service/impl/GetOperationTypeImpl.java

```diff
@@ -0,0 +1,18 @@
+package service.impl;
+
+import java.util.NoSuchElementException;
+import model.OperationType;
+import service.GetOperationType;
+
+public class GetOperationTypeImpl implements GetOperationType {
+    @Override
+    public OperationType checkAndGetOperationType(String operation) {
+        return switch (operation.toLowerCase()) {
+            case "b" -> OperationType.BALANCE;
+            case "s" -> OperationType.SUPPLY;
+            case "p" -> OperationType.PURCHASE;
+            case "r" -> OperationType.RETURN;
+            default -> throw new NoSuchElementException("OperationType " + operation + " no such");
+        };
```

Comment:
This method is tightly coupled with enum, so move it there. Replace switch expression with for-each iteration of enum's values(), this is a more flexible approach, you won't ever need to make changes there